### PR TITLE
Result.py: Add docstring regarding the case that line or column is None

### LINKED ISSUE #5310: Result.py: Add docstring regarding the case that line or column is None
The output generated by coala can have ```line``` being ```None``` or ```column``` being ```None```.

```column = None``` means there is an error with the whole line.
```line = None``` means the whole file. The reason behind it is that it's an unknown line means it can occur anywhere in the file.

All combinations with None values and numbers for line and column are allowed, except ```line=None``` and ```column=<some number>```

This should be clarified in the documentation.

(ref: http://api.coala.io/en/latest/coalib.results.html#module-coalib.results.Result)

### PR DESCRIPTION
**Pull request by ananyaarun**
Closes #5310

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/results/Result.py [hunk 1]
```diff
@@ -177,8 +177,13 @@ def from_values(cls,
             The related file.
         :param line:
             The first related line in the file. (First line is 1)
+            line = None, means the whole file.
+            That is an unknown line occurs anywhere in the file.
```
Comment (li-boxuan) [inline coalib/results/Result.py]: Remove this line, or make it clear.

File: coalib/results/Result.py [hunk 2]
```diff
@@ -177,8 +177,13 @@ def from_values(cls,
...
         :param column:
             The column indicating the first character. (First character is 1)
+            column = None, means there is an error with the whole line.
+            All combinations of None values are allowed for line and column,
+            Except line=None and column=<some number>
```
Comment (li-boxuan) [inline coalib/results/Result.py]: Add space around `=`.

**Comment by ananyaarun** [comment]
@li-boxuan , i have made the suggested changes ... kindly review

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**